src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java

```diff
@@ -87,19 +89,60 @@ public BattleshipModel placeComputerEasy(String level){
         return this;
     }
 
-    public BattleshipModel placeComputerHard(String level){
+    public BattleshipModel placeComputerHard(){
+        int i, //Iterator
+                j, //Iterator
+                k, //Iterator for shipName array of Strings
+                rowInt,
+                colInt;
+        boolean or, //Boolean for acquiring random orientation
+        overlapResult = false,
+        offBoard = false;
+        String orientation;
+        String[] shipName = new String[]{"aircraftcarrier", "battleship", "clipper", "submarine", "dinghy"};
 
-        if(level.equals("hard")){
-            System.out.println("in place computer(bsmod): " + level);
-            //placeComputerShips(sendLevel);
-        }
+        for(i=0, k=0; i<100; i++){
+            or = rand.nextBoolean();
+            rowInt = rand.nextInt(10) + 1;
+            colInt = rand.nextInt(10) + 1;
+            if(or)
```

Comment:
I assume this is the random placement of computer ships? looks good and likely works better than the original cycling idea.